### DISCUSSION TITLE
How to add "and" and "or" conditional operators

### DISCUSSION THREAD

**Discussion by davenewza** [google/cel-go#1055] [Q&A]
Is it possible to add `and` and `or` operators.  Ideally we'd replace `&&` and `||`, however an override would also be fine.  I've been digging through the library and it seems only functions can be added or overridden.  is this true?  If possible, please could you point me at an example.

**Comment by TristonianJones** [comment]
Hi @davenewza,

Generally, we prefer one way to do things with CEL, so we don't plan to support `and` and `or` at a language level. Currently, we use `or` as a function within `cel.OptionalTypes()` to indicate alternative expressions to produce a value, so introducing such operators would be quite painful as part of the core language since we'd have to support it syntactically and it might break some number of existing users.

-Tristan

**Comment by davenewza** [comment]
@TristonianJones thanks for the response.  Completely understand your case.  Best thing for us to do is rewrite the expression before parsing with CEL.
